superset/dashboards/dao.py

```diff
@@ -38,6 +38,20 @@ class DashboardDAO(BaseDAO):
     model_cls = Dashboard
     base_filter = DashboardFilter
 
+    @staticmethod
+    def get_dashboard(id_or_slug: str) -> Dashboard:
```

Comment:
nit: would prefer something like `find_by_id_or_slug` or `get_by_id_slug` we already know it's a dashboard